### PR TITLE
Add default loss function in_batch_sampled_softmax

### PR DESCRIPTION
**Pull request by ItsWayneYang**
Also, add misssing model.train in fit()

### CODE REVIEW

File: train_utils.py [hunk 1]
```diff
@@ -55,18 +179,35 @@ def fit(epochs,
...
+                *[loss_batch(model, loss_func, xb, yb, pairwise_similarity_func)
+                  for xb, yb in iterate_batch(test_data, batch_size)]
             )
+
+            evaluate(model, test_data, candidates, top_k, epoch)
         val_loss = np.sum(np.multiply(losses, nums)) / np.sum(nums)
 
-    print("Epoch {}: val_loss={}".format(epoch, val_loss))
\ No newline at end of file
+    print("Epoch {}: val_loss={}".format(epoch, val_loss))
```
Comment (adamziel) [inline train_utils.py:208]: It would be useful to print this on every epoch (or every nth)

File: train_utils.py [hunk 2]
```diff
@@ -55,18 +179,35 @@ def fit(epochs,
...
         for q1_batch, q2_batch in iterate_batch(train_data, batch_size):
-            loss_batch(model, loss_func, q1_batch, q2_batch, opt)
+            loss_batch(model, loss_func, q1_batch, q2_batch, pairwise_similarity_func, opt)
 
         model.eval()
         with torch.no_grad():
             losses, nums = zip(
-                *[loss_batch(model, loss_func, xb, yb) for xb, yb in iterate_batch(val_data, batch_size)]
+                *[loss_batch(model, loss_func, xb, yb, pairwise_similarity_func)
+                  for xb, yb in iterate_batch(test_data, batch_size)]
```
Comment (adamziel) [inline train_utils.py:202]: Nit: I think this should be called validation_data like in the older version of this file.

> Training Dataset: The sample of data used to fit the model.
> Validation Dataset: The sample of data used to provide an unbiased evaluation of a model fit on the training dataset while tuning model hyperparameters. The evaluation becomes more biased as skill on the validation dataset is incorporated into the model configuration.
> Test Dataset: The sample of data used to provide an unbiased evaluation of a final model fit on the training dataset.

https://machinelearningmastery.com/difference-test-validation-datasets/

Comment (adamziel) [reply train_utils.py:202]: Also, does it make sense to split validation data into mini batches vs evaluating on all the data at once?

Comment (adamziel) [reply train_utils.py:202]: To answer my own question, it does:
> In-batch precision@1 (accuracy computed over each row of the similarity matrix M)

**Comment by adamziel** [comment]
This looks great 👍
